scipy/stats/tests/gen_studentized_range_mpmath_ref.py

```diff
@@ -0,0 +1,186 @@
+# To run this script, run
+# `python gen_studentized_range_mpmath_ref.py`
+# in the "scipy/stats/tests/" directory
+
+# This script generates a JSON file "./data/studentized_range_mpmath_ref.json"
+# that is used to compare the accuracy of `studentized_range` functions against
+# precise (20 DOP) results generated using `mpmath`.
+
+# Note: I would have prefered to use pickle rather than JSON, but -
+# due to security concerns - decided against it.
+
+from collections import namedtuple
+import json
+import time
+
+from mpmath import gamma, pi, sqrt, quad, inf, mpf, mp, exp
+from mpmath import npdf as phi
+from mpmath import ncdf as Phi
+
+results_filepath = "data/studentized_range_mpmath_ref.json"
+
+MPResult = namedtuple("MPResult", ["src_case", "mp_result"])
+
+CdfCase = namedtuple("CdfCase",
+                     ["q", "k", "v", "expected_atol", "expected_rtol"])
+
+MomentCase = namedtuple("MomentCase",
+                        ["m", "k", "v", "expected_atol", "expected_rtol"])
+
+general_atol = 1e-11
+general_rtol = 1e-11
+mp.dps = 24
+
+cdf_cases = [
+    CdfCase(0.1, 3, 10, general_atol, general_rtol),
+    CdfCase(1, 3, 10, general_atol, general_rtol),
+    CdfCase(10, 3, 10, general_atol, general_rtol),
+
+    CdfCase(0.1, 10, 10, general_atol, general_rtol),
+    CdfCase(1, 10, 10, general_atol, general_rtol),
+    CdfCase(10, 10, 10, general_atol, general_rtol),
+
+    CdfCase(0.1, 3, 100, general_atol, general_rtol),
+    CdfCase(1, 3, 100, general_atol, general_rtol),
+    CdfCase(10, 3, 100, general_atol, general_rtol),
+
+    CdfCase(0.1, 10, 100, general_atol, general_rtol),
+    CdfCase(1, 10, 100, general_atol, general_rtol),
+    CdfCase(10, 10, 100, general_atol, general_rtol),
+
+    CdfCase(1, 3, 1000, general_atol, general_rtol),
+    CdfCase(10, 3, 1000, 1e-10, general_rtol),
+]
+
+# PDF can use same cases as CDF
+pdf_cases = cdf_cases
+
+mom_atol, mom_rtol = 1e-9, 1e-9
+# These are EXTREMELY slow - Multiple days each in worst case.
+moment_cases = [
+    MomentCase(1, 3, 10, mom_atol, mom_rtol),
+    MomentCase(2, 3, 10, mom_atol, mom_rtol),
+    MomentCase(3, 3, 10, mom_atol, mom_rtol),
+    MomentCase(4, 3, 10, mom_atol, mom_rtol)
+]
+
+
+def to_dict(named_tuple):
+    return dict(named_tuple._asdict())
+
+
+def mp_res_to_dict(mp_result):
+    """Formats an MPResult namedtuple into a dict for JSON dumping"""
+    return {
+        "src_case": to_dict(mp_result.src_case),
+
+        # np assert can't handle mpf, so take the accuracy hit here.
+        "mp_result": float(mp_result.mp_result)
+    }
+
+
+def cdf_mp(q, k, nu):
+    """Calculates the CDF integrand for the given q, k, nu vals"""
```

Comment:
It calculates the CDF, not just the integrand. 
Also, according to [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html):
![image](https://user-images.githubusercontent.com/6570539/117400428-73cd1380-aeb7-11eb-9fac-1e03b9029824.png)

So how about "Direct implementation of studentized range CDF"?

Similar for other functions. I'd also put a reference in this file. Probably point to Wikipedia, as this appears to be written in the same sort of order.